### PR TITLE
Update to node 14, update dependencies

### PR DESCRIPTION
**Pull request by mpcsh**
I've been noticing some `npm` warnings; this PR addresses them.
- `node` updated from v12 to v14 (latest is 14.16.1 as of now)
- Ran `npm audit fix` and `npm update` (these do not cross semver major boundaries)

Things that'd be nice to do:
- Address `npm outdated` output:
```
❯ npm outdated
Package                 Current  Wanted  Latest  Location
autoprefixer              9.8.6   9.8.6  10.2.5  selfdefined
concurrently              5.3.0   5.3.0   6.0.2  selfdefined
eslint                    6.8.0   6.8.0  7.25.0  selfdefined
eslint-config-prettier   6.15.0  6.15.0   8.3.0  selfdefined
husky                     4.3.8   4.3.8   6.0.0  selfdefined
markdown-it-anchor        5.3.0   5.3.0   7.1.0  selfdefined
markdownlint-cli         0.23.2  0.23.2  0.27.1  selfdefined
prettier                 1.19.1  1.19.1   2.2.1  selfdefined
```
- Address warnings on `npm run serve`:
```
[1] DEPRECATION WARNING: $hue: Passing a unit other than deg (120s) is deprecated.
[1] 
[1] To preserve current behavior: $hue / 1s
[1] 
[1] See https://sass-lang.com/d/color-units
[1] 
[1]   ╷
[1] 7 │   dark-green: hsl(120s, 100%, 15%),
[1]   │               ^^^^^^^^^^^^^^^^^^^^
[1]   ╵
[1]     assets/css/abstracts/_variables.scss 7:15  @import
[1]     assets/css/base.scss 4:9                   root stylesheet
[1] 
[1] WARNING: You probably don't mean to use the color value red in interpolation here.
[1] It may end up represented as red, which will likely produce invalid CSS.
[1] Always quote color names when using them as strings or map keys (for example, "red").
[1] If you really want to use the color value here, use '"" + $name'.
[1] 
[1]   ╷
[1] 4 │     --clr-#{$name}: #{$value};
[1]   │             ^^^^^
[1]   ╵
[1]     assets/css/base/_custom-properties.scss 4:13  raw-color-names()
[1]     assets/css/base/_custom-properties.scss 51:3  @import
[1]     assets/css/base.scss 12:9                     root stylesheet
```
⚠️ _Note: this output is incomplete and abridged._

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1 +0,0 @@
-nodejs 12.16.1
```
Comment (mpcsh) [inline .tool-versions:1]: This file is used by [`asdf`](https://asdf-vm.com), and was committed as part of a [larger commit](https://github.com/selfdefined/web-app/commit/beb2f5db69d81fffa92b294f5663624f12783de0) (likely due to a stray `git add .`). `asdf` is a lovely tool, but I don't think it's an actual build dependency of this app. As such, I propose that this file be removed; users of `asdf` (such as myself!) can create a [global `.gitignore`](https://davidwalsh.name/global-gitignore) which prevents this file from being committed to upstream projects.

**Comment by tatianamac** [comment]
@mpcsh This looks awesome! 

For the two "nice to haves", they sound good to me. What would you need from me to add these to this PR (or would you want to add these to this PR?)

**Review by tatianamac — APPROVED** [review]

**Comment by mpcsh** [comment]
I think the two nice-to-haves should probably be their own PRs. for `npm outdated`, major semver upgrades can introduce breaking changes, so we should probably open up a new branch and introduce them one at a time. for those CSS warnings, I think I'd need to pair on them. will DM 😁

**Comment by ovlb** [comment]
@mpcsh Could you also update the `.nvmrc` file in a future PR?

**Comment by mpcsh** [comment]
Oh, I think we should probably remove it for the same reason as with [this file](https://github.com/selfdefined/web-app/pull/404#discussion_r624218671) (i.e., we shouldn't commit tool versioning configs, that should be up to folks to manage as they see fit)

**Pull request merged**